src/OrchardCore.Modules/OrchardCore.Lists/Migrations.cs

```diff
@@ -53,5 +53,38 @@ public int UpdateFrom2()
 
             return 3;
         }
+
+        public int UpdateFrom3()
+        {
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .AddColumn<string>("ListContentType")
+            );
+
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .AddColumn<bool>("Published")
+            );
+
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .AddColumn<bool>("Latest")
+            );
+
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .AddColumn<string>("DisplayText")
+            );
+
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .DropIndex("IDX_ContainedPartIndex_DocumentId")
+            );
+
+            SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
+                .CreateIndex("IDX_ContainedPartIndex_DocumentId",
+                    "DocumentId",
+                    "ListContentItemId",
+                    "Order",
+                    "ListContentType")
+            );
```

Comment:
@jtkech from SQL benefit perspective, this is not a good idea. Creating indexes like that adds no value. SQL RDBMS like sql server and MySQL utilize indexes with columns left to right. That means if you have index that has columns `A`, `B`, `C` yet only column `B` and `C` appear in the query, then that index won't even be used. Also, there isn't much benefits to index a Boolean column unless you have a valid reason to do so. Same with the text field, unless you are planning query again that column with query like "search for this%" it's not really useful. I would also not include `Id` since that is a primary key and it isn't used by joins along with clause. 

Is there a benefit that I am missing that you are seeing?

Reply:
Yes, this is not so simple and it may depend on providers, I don't remember all details but we already did many tests mainly on Sql Server to find good compromises. As I remember here we use the full index pattern meaning only one index including all columns, yes when we do a joined query this is the `DocumentId` that is first used and then found in this only one index, then if it finds all what it needs the index is used.

> Also, there isn't much benefits to index a Boolean column unless you have a valid reason to do so

Retrieve in the front end all Published item Ids and DisplayText belonging to a given container. Yes I still hesitate for DisplayText which is longer, but I never hesitate for these light boolean fields.

> I would also not include Id since that is a primary key and it isn't used by joins

Yes but the Id is returned when we do a direct `.QueryIndex()`, not a joined query, so and only in this case it would never use this non clustered index if the Id is not also included, I did these kind of tests.

I already used this with custom index tables, what I also do is to populate caches from index tables  (using VolatileDocumentManager) by using `.QueryIndex()` and on which I can filter e.g. only Published items.

---

Hmm, as I already suggested maybe better to just not udpdate this index in this PR, as it can be done in a separate one (or never as on my side for collections I often used custom index tables, we even index BagPart items to quickly retrieve Pocos for a given parentId), otherwise it will never end ;)

Reply:
I don't buy that this is a good index "unless you are writing a huge query against this table

``` 
SchemaBuilder.AlterIndexTable<ContainedPartIndex>(table => table
            .CreateIndex("IDX_ContainedPartIndex_DocumentId",
                "Id",            // Only here, not in the Index Table and index provider.
                "DocumentId",
                "ContentItemId", // <= Easy to add this one, also in the index Table and provider.
                "ListContentItemId",
                "DisplayText",
                "ListContentType",
                "Order",
                "Published",
                "Latest")
        );
```

Reply:
Okay no problem, so maybe just remove all the changes related to this index table, then I will approve as you said that you already retried it, but again no problem as you want ;)

Reply:
@jtkech i'll make the other changes you requested in the morning. 

I don't mind leaving the index changes as it currently sits. Unless you want me to undo the index altogether.  I think the index as it is on the PR is beneficial.